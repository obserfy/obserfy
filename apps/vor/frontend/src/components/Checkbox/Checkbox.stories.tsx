import { Meta, Story } from "@storybook/react"
import Checkbox, { CheckboxProps } from "./Checkbox"

export default {
  title: "Core/Checkbox",
  component: Checkbox,
  parameters: {
    componentSubtitle: "Just a simple Checkbox",
  },
} as Meta

const Template: Story<CheckboxProps> = (args) => <Checkbox {...args} />

export const Default = Template.bind({})
Default.args = {}
