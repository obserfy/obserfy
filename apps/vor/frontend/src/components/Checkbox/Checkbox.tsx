import { FC } from "react"
import { ThemeUIStyleObject, Checkbox as BaseCheckbox, Label } from "theme-ui"
import { Trans } from "@lingui/macro"

export interface CheckboxProps {
  label: string
  onChange?: (value: boolean) => void
  checked?: boolean
  containerSx?: ThemeUIStyleObject
  defaultChecked?: boolean
  disabled?: boolean
}
const Checkbox: FC<CheckboxProps> = ({
  defaultChecked,
  containerSx,
  onChange,
  checked,
  label,
  disabled,
}) => (
  <Label sx={{ display: "flex", alignItems: "center", ...containerSx }}>
    <BaseCheckbox
      defaultChecked={defaultChecked}
      onChange={(e) => onChange?.(e.target.checked)}
      checked={checked}
      disabled={disabled}
    />
    <Trans id={label} />
  </Label>
)

export default Checkbox
