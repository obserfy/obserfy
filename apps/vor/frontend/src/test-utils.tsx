/* eslint-disable import/no-extraneous-dependencies */
import { FC, ReactElement } from "react"
import { QueryClient, QueryClientProvider } from "react-query"
import { ThemeProvider, Box, Theme } from "theme-ui"
import { render, RenderOptions, RenderResult } from "@testing-library/react"
import {
  createHistory,
  createMemorySource,
  LocationProvider,
} from "@reach/router"
import { I18nProvider } from "@lingui/react"
import { i18n } from "@lingui/core"
import { en } from "make-plural/plurals"
import theme from "./gatsby-plugin-theme-ui"
// @ts-ignore
import enCatalog from "../i18n/lingui/en/messages.js"

// any used here because the catalogs are generated b lingui
// not our responsibility
i18n.loadLocaleData("en", { plurals: en })
i18n.load("en", enCatalog.messages as any)
i18n.activate("en")

const queryClient = new QueryClient()
const history = createHistory(createMemorySource("/"))

const AllProviders: FC = ({ children }) => {
  return (
    <I18nProvider i18n={i18n}>
      <QueryClientProvider client={queryClient}>
        <LocationProvider history={history}>
          <ThemeProvider theme={theme as Theme}>
            <Box sx={{ fontSize: [16, 20] }}>{children}</Box>
          </ThemeProvider>
        </LocationProvider>
      </QueryClientProvider>
    </I18nProvider>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllProviders, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
