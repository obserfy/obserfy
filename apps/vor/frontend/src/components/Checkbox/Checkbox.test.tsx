import { render } from "../../test-utils"
import Checkbox from "./Checkbox"

describe("Checkbox", () => {
  it("should render correctly", () => {
    const { container } = render(<Checkbox label="just at test" />)
    expect(container).toMatchSnapshot()
  })
})
