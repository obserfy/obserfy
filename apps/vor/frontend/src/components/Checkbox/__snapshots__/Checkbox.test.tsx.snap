// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 16px;
}

@media screen and (min-width: 40em) {
  .emotion-0 {
    font-size: 20px;
  }
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: var(--theme-ui-colors-textMediumEmphasis);
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-size: 14px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  min-width: -webkit-min-content;
  min-width: -moz-min-content;
  min-width: min-content;
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  position: absolute;
  opacity: 0;
  z-index: -1;
  width: 1px;
  height: 1px;
  overflow: hidden;
}

.emotion-4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: none;
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-right: 8px;
  border-radius: 4px;
  color: var(--theme-ui-colors-gray);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

input:checked~.emotion-4 {
  display: block;
}

input:checked~.emotion-4 {
  color: var(--theme-ui-colors-primary);
}

input:focus~.emotion-4 {
  color: var(--theme-ui-colors-primary);
  background-color: var(--theme-ui-colors-highlight);
}

.emotion-5 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-right: 8px;
  border-radius: 4px;
  color: var(--theme-ui-colors-gray);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

input:checked~.emotion-5 {
  display: none;
}

input:checked~.emotion-5 {
  color: var(--theme-ui-colors-primary);
}

input:focus~.emotion-5 {
  color: var(--theme-ui-colors-primary);
  background-color: var(--theme-ui-colors-highlight);
}

<div>
  <div
    class="emotion-0"
  >
    <label
      class="emotion-1"
    >
      <div
        class="emotion-2"
      >
        <input
          class="emotion-3"
          type="checkbox"
        />
        <svg
          aria-hidden="true"
          class="emotion-4"
          fill="currentcolor"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="emotion-5"
          fill="currentcolor"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </svg>
      </div>
      just at test
    </label>
  </div>
</div>
`;
